test/steps/process.go

```diff
@@ -41,6 +44,49 @@ func (data *Data) startProcessOnService(processName, serviceName string, body *m
 	return nil
 }
 
+func (data *Data) startProcessOnServiceWithinMinutes(processName, serviceName string, timeoutInMin int, body *messages.PickleStepArgument_PickleDocString) error {
+	routeURI, err := framework.WaitAndRetrieveRouteURI(data.Namespace, serviceName)
+	if err != nil {
+		return err
+	}
+
+	return framework.WaitForOnOpenshift(data.Namespace, fmt.Sprintf("Service %s is not available yet", serviceName), timeoutInMin,
+		func() (bool, error) {
+			err = framework.StartProcess(data.Namespace, routeURI, processName, body.GetMediaType(), body.GetContent())
+			if err != nil {
+				return false, err
+			}
+
+			return true, nil
+		})
+}
+
+func (data *Data) serviceWithProcessNameIsAvailable(serviceName, processName string) error {
+	routeURI, err := framework.WaitAndRetrieveRouteURI(data.Namespace, serviceName)
+	if err != nil {
+		return err
+	}
+
+	_, err = framework.GetProcessInstances(data.Namespace, routeURI, processName)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (data *Data) serviceWithProcessNameIsAvailableWithinMinutes(serviceName, processName string, timeoutInMin int) error {
+	return framework.WaitForOnOpenshift(data.Namespace, fmt.Sprintf("Process %s is not available yet", processName), timeoutInMin,
+		func() (bool, error) {
+			err := data.serviceWithProcessNameIsAvailable(serviceName, processName)
```

Comment:
Here we are going to get each time the route URI (and wait for it). is that expected ?
I think we should get it once before the wait and then just call the `framework.GetProcessInstances` method

Reply:
I wanted to reuse the first method and thought getting each time the route it was not a big issue...
Changed.